base/dates/periods.jl

```diff
@@ -439,8 +438,13 @@ let vmax = typemax(Int64) ÷ 12, vmin = typemin(Int64) ÷ 12
 end
 Base.convert(::Type{Year}, x::Month) = Year(divexact(value(x), 12))
 Base.promote_rule(::Type{Year}, ::Type{Month}) = Month
-(==){T<:OtherPeriod, S<:OtherPeriod}(x::T, y::S) = (==)(promote(x, y)...)
-Base.isless{T<:OtherPeriod, S<:OtherPeriod}(x::T, y::S) = isless(promote(x, y)...)
+
+# disallow comparing fixed to other periods
+(==){T<:FixedPeriod, S<:OtherPeriod}(x::T, y::S) = throw(MethodError(==, (T, S)))
+(==){T<:OtherPeriod, S<:FixedPeriod}(x::T, y::S) = throw(MethodError(==, (T, S)))
```

Comment:
The `MethodError` should contain the arguments, not their types (when possible).